### ISSUE TITLE
How to calculate "cncf" column?

### ISSUE THREAD

**Issue by leap-ahead225** [KarchinLab/pictograph#21]
Thank you for the great tool.

I think when I run this software I need information on the cncf columns, how can I calculate this?
Sorry for the basic question.

**Comment by gbnci** [comment]
Hello: Could you please give a little bit more information about the "cncf"? I am not quite sure what it represents. I may miss it somewhere but just could not find it anywhere. Thanks for your help.

**Comment by jiaying2508** [comment]
Hi! cncf stands for copy-number cellular fraction, i.e. the fraction of cells that have the copy number alteration. The copy number alteration for a mutation is the one that overlaps with the mutation. If for a given mutation there is no copy number alteration, you can put cncf of 0. Otherwise you should put the fraction. cncf can be obtained from copy number callers that enable subclonal copy number detection. One tool commonly used is FACETS, developed by MSKCC.

**Comment by mhoang22** [comment]
Hi pictograph developers,

Im interested in using the tool and would love to have some more clarification on cncf.

In the first row in example 1, cncf is noted as 0.8. 
https://github.com/KarchinLab/pictograph/blob/master/inst/extdata/example1_snv.csv
Does it mean that specific mutation is on a region with copy number alteration, and that specific copy number alteration event is seen in 80% of the cells in the tumor?

Also in the previous comment, @jiaying2508 mentioned that a tool you commonly used to infer cncf is FACETS by SMKCC.
Could you talk more about this process?
Would the 'cncf' be the output of 
+ running command 'emncf' mskcc/facets
+ or running annotate-maf-wrapper.R in mskcc/facets-suite 
? 
When I look into running mskcc/facets, it looks like this estimate fraction and allele specific copy number using a vcf with common variants in a **population** (eg. vcf from dbsnp for 1000 genome) , **not** vcf of variants from my patient tumors. 
https://github.com/mskcc/facets/issues/66
**So I wonder how one finds the cncf for somatic variants unique in my patient tumor** (which are not in dbsnp) ?
Does pictograph only support snp in population level?
Or do I run facets on the vcf with my patient tumor variants only, and use the 'cf.em' in fit$cncf (output of emncf in mskcc/facet) as the cncf for pictograph?

Thanks in advance for your time and help!

**Comment by jiaying2508** [comment]
@mhoang22 glad to hear that you are interested in our tool!

> Does it mean that specific mutation is on a region with copy number alteration, and that specific copy number alteration event is seen in 80% of the cells in the tumor?

Yes, that's correct.

> Also in the previous comment, mentioned that a tool you commonly used to infer cncf is FACETS by SMKCC.
Could you talk more about this process?

To run FACETS, you can refer to their GitHub page https://github.com/mskcc/facets. You would need to run the tool for your specific sample, so you get the copy number profile **specific** for your sample. You will then identify where the mutation is located and find the corresponding copy number cellular fraction from the FACETS output, from the function emcncf() and column named "cf.em" in fit$cncf. 

Let me know if this makes sense.

**Comment by mhoang22** [comment]
Hi @jiaying2508 ,

Thanks for the speedy response. That indeed answered my question.

Just to make sure that I'm understanding this correctly, say, I have 3 WES samples: (1) normal, (2) tumor-timepoint1, (3) tumor-timepoint2.
The following is my plan for performing the analysis, could you comment if anything doesnt look right?

+ Step 1: run `snp-pile` up for each normal-tumor bam pair
( Effectively, run snp-pile for 2 comparisons: Normal vs tumor-timepoint1 and normal vs tumor-timepoint2, using input vcf as list of all somatic variants seen in 2 tumors.) 
(**Or should I use list of germline + somatic variants in 2 tumors?**)

+ Step 2: run facets emncf for each of the 2 matrices from 2 comparisons above. 
Take note of the ‘cf.em’ for each variant in the output. 

+ Step 3: Prep input for pictograph.
**For pictograph input, should I only list 2 samples: tumor-timepoint1, tumor-timepoint2?** (or should I actually list 3 samples: normal, tumor-timepoint1, tumor-timepoint2 ?) 

For pictograph input, the first mutation value in example_1 is ‘chr1-57-clone1’.  
https://github.com/KarchinLab/pictograph/blob/master/inst/extdata/example1_snv_with_purity.csv
**Does this mean that there’s a step which assign clone identity to each variant prior to the pictograph ‘mcmcMain’ step?**
Or would pictograph still work if I dont assign clone to each variant, and have the 'mutation' value just being the variant coordinate?

+ Step 4: run pictograph ‘mcmcMain’ (or run each individual function in the tool).

Best,
My

**Comment by jiaying2508** [comment]
@mhoang22 

> Step 1: run snp-pile up for each normal-tumor bam pair
> ( Effectively, run snp-pile for 2 comparisons: Normal vs tumor-timepoint1 and normal vs tumor-timepoint2, using input vcf as list of all somatic variants seen in 2 tumors.)

Yes, you would need to run FACETS for each tumor-normal pair. I am not sure what you mean by "using input vcf as list of all somatic variants seen in 2 tumors.", but you should refer to FACETS GitHub about running the tool.

> Step 2: run facets emncf for each of the 2 matrices from 2 comparisons above.
> Take note of the ‘cf.em’ for each variant in the output.

That's correct.

> Step 3: Prep input for pictograph.
> For pictograph input, should I only list 2 samples: tumor-timepoint1, tumor-timepoint2? (or should I actually list 3 samples: normal, tumor-timepoint1, tumor-timepoint2 ?)

You should only list the two tumor samples.

> Step 4: run pictograph ‘mcmcMain’ (or run each individual function in the tool).

Yes.

**Comment by mhoang22** [comment]
@jiaying2508 

thanks so much for your reply.

Regarding the vcf: I was initially confused since facets uses a dbsnp vcf as input, and I wasnt sure how that would help identify the cncf for my sample somatic variants. I just reread your comment again and realize that the output of facets is 'copy number profile specific for (my) sample', and that I will need to locate where my somatic variants are in this copy number profile to infer each somatic variant respective cncf. Sorry I didnt get your reply the first time around, the course of action is clear to me now, thanks again!

Also reiterating 1 question again in case you didnt see this: 

> For pictograph input, the first mutation value in example_1 is ‘chr1-57-clone1’.
> https://github.com/KarchinLab/pictograph/blob/master/inst/extdata/example1_snv_with_purity.csv
> **Is the '-clone1' (clone number) part required?** 
> **Does this mean that there’s a step which assign clone identity to each variant prior to the pictograph ‘mcmcMain’ step?**

**Comment by jiaying2508** [comment]
@mhoang22 

> Is the '-clone1' (clone number) part required?

No, it's not required. Sorry for the confusion, I should have named the mutations better. The mutation column is just a name you provide for the mutation.

**Comment by mhoang22** [comment]
gotcha. thanks so much for the clarification!
